### PR TITLE
init interaction to contract

### PR DESCRIPTION
**Pull request by chan-aa**

### CODE REVIEW

File: pypact/test/test_interaction.py [hunk 1]
```diff
@@ -3,19 +3,18 @@
...
 TEST_RESPONSE = {
-    'status': 200,
-    'headers': {'Custom-Header': 'value'},
-    'body': {'key': 'value'}}
+    "status": 200,
+    "headers": {"Custom-Header": "value"},
+    "body": {"key": "value"}}
 
 
 @pytest.fixture
```
Comment (luyun-aa) [inline pypact/test/test_interaction.py]: We won't use pytest

File: pypact/service.py [hunk 2]
```diff
@@ -9,12 +9,14 @@ class MockService(object):
     def __init__(self, consumer, provider, port, interaction_builder=None):
         self.consumer = consumer
         self.provider = provider
+        self.
```
Comment (luyun-aa) [inline pypact/service.py]: ...

**Comment by chan-aa** [comment]
@luyun-aa please help to review.

File: pypact/interaction.py [hunk 3]
```diff
@@ -3,13 +3,15 @@ class Interaction(object):
...
-    def __init__(self, add_method):
-        self.add_method = add_method
+    def __init__(self, add_method=None):
+        # self.add_method = add_method
 
         self.provider_state = None
         self.description = None
         self.request = None
         self.response = None
+        self.all_method = add_method
```
Comment (luyun-aa) [inline pypact/interaction.py]: `all_method` vs `add_method`?

File: pypact/test/test_contract_factory.py [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+                    "version": "1.0.0"
+                },
+                "pact-pypact": {
+                    "version": "1.0.0"
+                }
+            }
+        }
+        print ">>>>>"
+        print actually_contract
+        print expected_contract
```
Comment (luyun-aa) [inline pypact/test/test_contract_factory.py]: Please remove the `print`s

File: pypact/test/test_mock_service.py [hunk 5]
```diff
@@ -1,127 +1,207 @@
...
+
+from pypact.service import MockService
+
+
+class TestMockServer(unittest.TestCase):
+    def test_mock_service_functional_json(self):
+        service = MockService(port=1234)
+
+        service.given("Test").upon_receiving("a request").with_request(
+                method="get",
```
Comment (luyun-aa) [inline pypact/test/test_mock_service.py]: Please stick to 4 space indentation

File: pypact/test_old/test_mock_service.py [hunk 6]
```diff
@@ -0,0 +1,154 @@
+import pytest
```
Comment (luyun-aa) [inline pypact/test_old/test_mock_service.py]: just remove the old tests entirely?

File: pypact/test/utils.py [hunk 7]
```diff
@@ -0,0 +1,8 @@
+
```
Comment (luyun-aa) [inline pypact/test/utils.py]: don't need this code anymore

**Pull request merged**